### PR TITLE
5.3 registration

### PR DESCRIPTION
**Pull request by Amnander**

### CODE REVIEW

File: src/main/java/com/ecommerce/website/configuration/AuthConfig.java [hunk 1]
```diff
@@ -64,6 +64,6 @@ AuthenticationManager authenticationManager(AuthenticationConfiguration authenti
 
     @Bean
     PasswordEncoder passwordEncoder() {
-        return new BCryptPasswordEncoder();
+        return new BCryptPasswordEncoder(12);
```
Comment (ZigmundsR) [inline src/main/java/com/ecommerce/website/configuration/AuthConfig.java:67]: Why this is needed?

File: src/main/java/com/ecommerce/website/configuration/AuthConfig.java [hunk 2]
```diff
@@ -35,7 +35,7 @@ SecurityFilterChain securityFilterChain(HttpSecurity httpSecurity) throws Except
                 .sessionManagement(session -> session.sessionCreationPolicy(SessionCreationPolicy.IF_REQUIRED))
                 .authorizeHttpRequests(authorize -> authorize
                         .requestMatchers("/").permitAll()
-                        .requestMatchers("/images/**").permitAll()  // Allow access to static images
+                        .requestMatchers("/images/**").permitAll()
```
Comment (ZigmundsR) [inline src/main/java/com/ecommerce/website/configuration/AuthConfig.java:38]: I think it was okay with the comment?

File: src/main/java/com/ecommerce/website/configuration/AuthConfig.java (hunk 1 above)
Comment (Amnander) [reply src/main/java/com/ecommerce/website/configuration/AuthConfig.java:67]: Its a strength parameter for the Bcrypt algo. If you want i can remove it?

File: src/main/java/com/ecommerce/website/service/UserService.java [hunk 3]
```diff
@@ -25,11 +25,13 @@ public UserService(UserRepository repository) {
...
     @Override
-    public UserDetails loadUserByUsername(String username) {
+    public UserDetails loadUserByUsername(String username)
+    {
         return repository.findByLogin(username).orElseThrow(() -> new UsernameNotFoundException("user not found"));
     }
 
-    public User findByUsername(String username) {
+    public User findByUsername(String username)
+    {
```
Comment (ZigmundsR) [inline src/main/java/com/ecommerce/website/service/UserService.java:34]: By default formatting I don't think it does this? You think this is better? (personally, I don't like this look for code)

File: src/main/java/com/ecommerce/website/configuration/AuthConfig.java (hunk 2 above)
Comment (Amnander) [reply src/main/java/com/ecommerce/website/configuration/AuthConfig.java:38]: Okay i'll put that comment back.

File: src/main/java/com/ecommerce/website/service/UserService.java (hunk 3 above)
Comment (Amnander) [reply src/main/java/com/ecommerce/website/service/UserService.java:34]: Okay i'll fix it.

**Pull request merged**